subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaIntegTest.groovy

```diff
@@ -612,6 +612,57 @@ class MavenPublishJavaIntegTest extends AbstractMavenPublishIntegTest {
 
     }
 
+    @Unroll
+    def "can publish #thing component with capabilities"() {
+        given:
+        createBuildScripts("""
+            dependencies {
+                capabilities {
+                    capability('org:cap') {
+                        providedBy 'org:foo'
+                    }
```

Comment:
What if only some versions of 'org:foo' provide this capability?

Reply:
That's not supported: in the initial PR that added support for capabilities, and as a result of the spike, we chose not to do this and state that capabilities are valid independently of the version number. They are always valid, whatever the version, whatever the variant.